custom_components/isy994/config_flow.py

```diff
@@ -0,0 +1,126 @@
+"""Config flow for Universal Devices ISY994 integration."""
+from functools import partial
+import logging
+from urllib.parse import urlparse
+
+from pyisy import ISY
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+
+from .const import (
+    CONF_ENABLE_CLIMATE,
+    CONF_IGNORE_STRING,
+    CONF_SENSOR_STRING,
+    CONF_TLS_VER,
+    DEFAULT_IGNORE_STRING,
+    DEFAULT_SENSOR_STRING,
+)
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_TLS_VER): vol.Coerce(float),
+        vol.Optional(CONF_IGNORE_STRING, default=DEFAULT_IGNORE_STRING): str,
+        vol.Optional(CONF_SENSOR_STRING, default=DEFAULT_SENSOR_STRING): str,
+        vol.Optional(CONF_ENABLE_CLIMATE, default=True): bool
+        # Variables require yaml
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    user = data[CONF_USERNAME]
+    password = data[CONF_PASSWORD]
+    host = urlparse(data[CONF_HOST])
+    tls_version = data.get(CONF_TLS_VER)
+
+    if host.scheme == "http":
+        https = False
+        port = host.port or 80
+    elif host.scheme == "https":
+        https = True
+        port = host.port or 443
+    else:
+        _LOGGER.error("isy994 host value in configuration is invalid")
+        raise InvalidHost
+
+    # Connect to ISY controller.
+    isy = await hass.async_add_executor_job(
+        partial(
+            ISY,
+            host.hostname,
+            port,
+            username=user,
+            password=password,
+            use_https=https,
+            tls_ver=tls_version,
+            log=_LOGGER,
+        )
+    )
+    if not isy.connected:
+        raise InvalidAuth
+
+    # Return info that you want to store in the config entry.
+    return {"title": host.hostname, "uuid": isy.configuration["uuid"]}
```

Comment:
You can use the name now as of `dev136`:
```python
return {"title": f"{isy.configuration['name']} ({host.hostname})", "uuid": isy.configuration["uuid"]}
```

Reply:
nice.

done